django/core/management/base.py

```diff
@@ -289,28 +291,42 @@ def create_parser(self, prog_name, subcommand, **kwargs):
         parser.add_argument(
             '-v', '--verbosity', default=1,
             type=int, choices=[0, 1, 2, 3],
-            help='Verbosity level; 0=minimal output, 1=normal output, 2=verbose output, 3=very verbose output',
+            help=argparse.SUPPRESS if 'verbosity' in self.suppressed_base_arguments else (
+                'Verbosity level; 0=minimal output, 1=normal output,'
+                '2=verbose output, 3=very verbose output'
+            )
         )
         parser.add_argument(
             '--settings',
-            help=(
+            help=argparse.SUPPRESS if 'settings' in self.suppressed_base_arguments else (
                 'The Python path to a settings module, e.g. '
                 '"myproject.settings.main". If this isn\'t provided, the '
                 'DJANGO_SETTINGS_MODULE environment variable will be used.'
-            ),
+            )
         )
         parser.add_argument(
             '--pythonpath',
-            help='A directory to add to the Python path, e.g. "/home/djangoprojects/myproject".',
+            help=argparse.SUPPRESS if 'pythonpath' in self.suppressed_base_arguments else (
+                'A directory to add to the Python path,'
+                'e.g. "/home/djangoprojects/myproject".'
+            )
+        )
+        parser.add_argument(
+            '--traceback', action='store_true',
+            help=argparse.SUPPRESS if 'traceback' in self.suppressed_base_arguments else (
+                'Raise on CommandError exceptions')
         )
-        parser.add_argument('--traceback', action='store_true', help='Raise on CommandError exceptions')
         parser.add_argument(
             '--no-color', action='store_true',
-            help="Don't colorize the command output.",
+            help=argparse.SUPPRESS if 'no-color' in self.suppressed_base_arguments else (
+                "Don't colorize the command output."
+            )
         )
         parser.add_argument(
             '--force-color', action='store_true',
-            help='Force colorization of the command output.',
+            help=argparse.SUPPRESS if 'force-color' in self.suppressed_base_arguments else (
+                'Force colorization of the command output.'
+            )
```

Comment:
I feel like this boiler-plate could be handled more nicely. For example, what about defining a function above that looks something like--

```python
def add_argument(parser, name, *args, help=None, **kwargs):
    if name in self.suppressed_base_arguments:
        help = argparse.SUPPRESS
    parser.add_argument(*args, help=help, **kwargs)
```

Then each `parser.add_argument(...)` would become `add_argument(parser, name, ...)`.

I also think it would be better if the convention were for the string in `suppressed_base_arguments` to match the first option string passed to `parser.add_argument()` (e.g. `--force-color` instead of `force-color`). I think it would be easier to remember. Also, if that were done, the name wouldn't have to be passed a second time, or manipulated in any way inside the helper function above before checking for membership in `self.suppressed_base_arguments`.

Reply:
That's clever idea @cjerdonek. 
I've noticed that passing help=None for the --version argument changes the output - no help text at all, where in the current implementation it sets "show program's version number and exit" as default,  despite help value is not passed at all?

I could hardcode that value for help if not suppressed, but I would prefer to know where it is defined?

Reply:
I've pushed a bit modified version of your concept and added relevant test, especially to check what happens with '--verbosity' using additional -v flag. All tests are passing and help displays both flags for verbosity.